demo/fake-api.js

```diff
@@ -0,0 +1,619 @@
+import {
+  createServer,
+  Model,
+  Response,
+} from "https://cdn.jsdelivr.net/npm/miragejs@0.1.47/+esm";
+import { Server } from "https://cdn.jsdelivr.net/npm/mock-socket@9.2.1/+esm";
+
+let wsUrl =
+  (window.location.protocol === "https:" ? "wss" : "ws") +
+  "://cryostat-demo.local:8181/fake";
+let wsServer = new Server(wsUrl);
+let websocket;
+wsServer.on("connection", (socket) => {
+  websocket = socket;
+  socket.on("message", (_) => {
+    socket.send(
+      JSON.stringify({
+        meta: {
+          category: "WsClientActivity",
+          type: {
+            type: "application",
+            subtype: "json",
+          },
+          serverTime: +Date.now(),
+        },
+        message: {
+          localhost: "connected",
+        },
+      })
+    );
+  });
+});
+
+createServer({
+  models: {
+    target: Model,
+    recording: Model,
+    archive: Model,
+    rule: Model,
+  },
+
+  seeds(server) {
+    server.create("target", {
+      alias: "Fake Target",
+      connectUrl: "http://fake-target.local:1234",
+      jvmId: "1234",
+      annotations: {
+        platform: { "io.cryostat.demo": "this-is-not-real" },
+        cryostat: { hello: "world" },
+      },
+    });
+  },
+
+  routes() {
+    this.timing = 0;
+
+    this.get("/health", () => ({
+      cryostatVersion: "v2.3.0-live-demo",
```

Comment:
Seems to causes parsing errors.

![Screenshot from 2023-05-09 17-08-18](https://github.com/cryostatio/cryostatio.github.io/assets/68053619/087d1d88-3b5b-4ba9-92d4-b2e45ca7db9a)

How about `v2.4.0-0-livedemo`?

Reply:
I was trying to play around with getting this version string either from the `bash build-demo.bash cryostat-v2.3` command, or from the `cryostat-web package.json`, but in the end this means having to patch the `fake-api.js` with the result, and that's pretty annoying for version control. If/when we do a better fake API server like we were talking about then this can probably be done better there with some other automation.